### DISCUSSION TITLE
How can I include the EOLN in a conditional formatting

### DISCUSSION THREAD

**Discussion by DavidThielen** [axuno/SmartFormat#376] [Q&A] [answered]
Hi all;

I have the following as part of my text for SmartFormat:
```
When: {Event.StartDateTime:M} {Event.StartDateTime:t} - {Event.EndDateTime:t}.

{Event.ShowAddress:cond:Where: {Event.Address}|}

{Event.ShowRemote:cond:{Event.Zoom.Description}: {Event.Zoom.Url}|}

Again, thank you for stepping up to help. This is how we win.
```
I also tried:
```
When: {Event.StartDateTime:M} {Event.StartDateTime:t} - {Event.EndDateTime:t}.

{Event.ShowAddress:cond:Where: {Event.Address}
|}
{Event.ShowRemote:cond:{Event.Zoom.Description}: {Event.Zoom.Url}
|}
Again, thank you for stepping up to help. This is how we win.
```
It does great on the cond only showing the line if ShowAddress is true. But if ShowAddress is false, I get 2 blank lines. And I want the lines spread out, with a blank line between each, if both conditions are true.

How can I accomplish this?

thanks - dave

**Comment by DavidThielen** [comment]
ps - this is a really useful library.

**Reply by axunonb** [reply, marked as answer]
Hi, currently I'm on vacation and limited to a mobile device.
I read `EOLN` as the carriage-return character. You can use it in the format string by using the `\n` text literal.

But in your code, the the blank lines come from the new lines outside of the placeholders. Put them in one line and use the `\n` as a text literal for a blank line.

Hope this helps.

**Reply by axunonb** [reply]
Without a working code sample it's a bit difficult to give a precise answer. What I mean is to put both placeholders on 1 line, and add then \n literal in those cases where you need it:
```CSharp
{Event.ShowAddress:cond:Where: {Event.Address}\n|}{Event.ShowRemote:cond:{Event.Zoom.Description}: {Event.Zoom.Url}\n|}
```
